etcdctl/command/util.go

```diff
@@ -286,10 +290,10 @@ func contextWithTotalTimeout(c *cli.Context) (context.Context, context.CancelFun
 // syncWithPeerAPI syncs cluster with peer API defined at
 // https://github.com/coreos/etcd/blob/v0.4.9/server/server.go#L311.
 // This exists for backward compatibility with etcd 0.4.x.
-func syncWithPeerAPI(c *cli.Context, ctx context.Context, knownPeers []string) ([]string, error) {
-	tr, err := getTransport(c)
-	if err != nil {
-		return nil, err
+func syncWithPeerAPI(c *cli.Context, ctx context.Context, knownPeers []string) (eps []string, trerr error, serr error) {
```

Comment:
I feel the error returned by this function are all connection error. Then only possible server error is that the returned eps cannot be parsed (not valid endpoints)